html_sanitizer.rst

```diff
@@ -0,0 +1,983 @@
+:method:`Symfony\\Component\\HtmlSanitizer\\HtmlSanitizer::sanitizeFor`
+method, you can instruct HTML sanitizer to customize this for the
+``<head>`` or a more specific HTML tag::
+
+    // tags not allowed in <head> will be removed
+    $safeInput = $htmlSanitizer->sanitizeFor('head', $userInput);
+
+    // encodes the returned HTML using HTML entities
+    $safeInput = $htmlSanitizer->sanitizeFor('title', $userInput);
+    $safeInput = $htmlSanitizer->sanitizeFor('textarea', $userInput);
+
+    // uses the <body> context, removing tags only allowed in <head>
+    $safeInput = $htmlSanitizer->sanitizeFor('body', $userInput);
+    $safeInput = $htmlSanitizer->sanitizeFor('section', $userInput);
+
+.. _html-sanitizer-configuration:
+
+Configuration
+-------------
+
+The behavior of the HTML sanitizer can be fully customized. This allows you
+to explicitly state which elements, attributes and even attribute values
+are allowed.
+
+You can do this by defining a new HTML sanitizer in the configuration:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/packages/html_sanitizer.yaml
+        framework:
+            html_sanitizer:
+                sanitizers:
+                    app.post_sanitizer:
+                        block_elements:
+                            - h1
+
+    .. code-block:: xml
+
+        <!-- config/packages/html_sanitizer.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xmlns:framework="http://symfony.com/schema/dic/symfony"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                https://symfony.com/schema/dic/services/services-1.0.xsd
+                http://symfony.com/schema/dic/symfony https://symfony.com/schema/dic/symfony/symfony-1.0.xsd">
+
+            <framework:config>
+                <framework:html-sanitizer>
+                    <framework:sanitizer name="app.post_sanitizer">
+                        <framework:block-element name="h1"/>
+                    </framework:sanitizer>
+                </framework:html-sanitizer>
+            </framework:config>
+        </container>
+
+    .. code-block:: php
+
+        // config/packages/framework.php
+        use Symfony\Config\FrameworkConfig;
+
+        return static function (FrameworkConfig $framework) {
+            $framework->htmlSanitizer()
+                ->sanitizer('app.post_sanitizer')
+                    ->blockElement('h1')
+            ;
+        };
+
+    .. code-block:: php-standalone
+
+        use Symfony\Component\HtmlSanitizer\HtmlSanitizer;
+        use Symfony\Component\HtmlSanitizer\HtmlSanitizerConfig;
+
+        $postSanitizer = new HtmlSanitizer(
+            (new HtmlSanitizerConfig())
+                ->blockElement('h1')
+        );
+
+This configuration defines a new ``app.post_sanitizer`` service. This service
```

Comment:
the service will be named `html_sanitizer.sanitizer.app.post_sanitizer` actually